pgjdbc/src/main/java/org/postgresql/jdbc/PgConnection.java

```diff
@@ -1425,7 +1432,9 @@ public boolean isValid(int timeout) throws SQLException {
         }
         return true;
       } finally {
-        setNetworkTimeout(null, savedNetworkTimeOut);
+        if (changedNetworkTimeout) {
```

Comment:
Do we really need `changedNetworkTimeout` can't we just compare the old to the new ?

Reply:
We'll need to compare the old to the current (`getNetworkTimeout()`), not to the new, and `getNetworkTimeout()` is not a simple getter, so maybe it's better to avoid calling it when we don't have to.

Reply:
right but you have the old as you got it above. So there is no need to create a flag. Just compare the new to the old here.

Reply:
The `newNetworkTimeout` is not always set as a new timeout (yeah, bad naming) in the code above - only when that not-so-easy-to-read condition is met. So if we compare the old to the new, we may end up calling `setNetworkTimeout()` without a reason.

Reply:
I see a way around this but at this point I'd be nit picking so just leave it the way it is.